src/clib/pioc_support.c

```diff
@@ -93,19 +94,57 @@ int PIOc_strerror(int pioerr, char *errmsg)
 int PIOc_set_log_level(int level)
 {
 #if PIO_ENABLE_LOGGING
-    char log_filename[NC_MAX_NAME];
-
     /* Set the log level. */
     pio_log_level = level;
+#endif /* PIO_ENABLE_LOGGING */
+    return PIO_NOERR;
+}
 
-    /* Create a filename with the rank in it. */
-    MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
-    sprintf(log_filename, "pio_log_%d.txt", my_rank);
+/** Initialize logging
+ * Open log file, if not opened yet, or
+ * increment ref count if already open
+ */
+void pio_init_logging(void )
+{
+#if PIO_ENABLE_LOGGING
+    char log_filename[NC_MAX_NAME];
+
+    if (!LOG_FILE)
+    {
+        /* Create a filename with the rank in it. */
+        MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
+        sprintf(log_filename, "pio_log_%d.txt", my_rank);
+
+        /* Open a file for this rank to log messages. */
+        LOG_FILE = fopen(log_filename, "w");
+
+        pio_log_ref_cnt = 1;
+    }
+    else
+    {
+        pio_log_ref_cnt++;
```

Comment:
If multiple components (or the same component multiple times) makes calls to PIOc_set_log_level, then I think it should change logging to the specified level for that task, and append to the existing log file. (And if you set it to different levels on the same task in different components, you get the last one set.)

That's not what will happen right now. At the moment, the file will be overwritten every time the log level changes, and that is clearly a bug. So let's fix that.

@rjacob the fist message logged (and this is very helpful) is the one saying that the log level has been set. This is useful because it means that the logging is working. There are also some functions that use logging that are called before pio_init. (The nc4 functions that can control netCDF-4 optimizations. Yes, I know they are a pain. But logging is still very useful there.)

So there is good reason to start the log files in the first call any task makes to PIOc_set_log_level().

The log files should be closed (if open) in PIOc_finalize(). It is a bug that they are not currently closed.